src/Features/LanguageServer/ProtocolUnitTests/Hover/EditorConfigHoverTests.cs

```diff
@@ -0,0 +1,355 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.EditorConfigSettings;
+using Microsoft.CodeAnalysis.CSharp.EditorConfigSettings;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Roslyn.Test.Utilities;
+using Xunit;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.UnitTests.Hover
+{
+    public class EditorConfigHoverTests : AbstractLanguageServerProtocolTests
+    {
+        private static readonly LSP.ClientCapabilities s_hoverCapabilities = new LSP.ClientCapabilities
+        {
+            TextDocument = new LSP.TextDocumentClientCapabilities
+            {
+                Hover = new LSP.HoverSetting
+                {
+                    ContentFormat = new LSP.MarkupKind[]
+                    {
+                        LSP.MarkupKind.Markdown
+                    }
+                }
+            }
+        };
+
+        [Fact]
+        public async Task TestShowHoverOnWhitespaceSettingName()
+        {
+            var markup = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"" FilePath=""z:\TestProject\Assembly1.csproj"">
+        <Document FilePath=""File1.cs"">
+class C { }
+        </Document>
+        <AnalyzerConfigDocument FilePath=""z:\TestProject\.editorconfig"">
+insert{|caret:_|}final_newline = true
+        </AnalyzerConfigDocument>
+    </Project>
+</Workspace>";
+
+            using var testLspServer = await CreateXmltestLspServerAndUpdateWithAnalyzerReferences(markup);
+            var expectedLocation = testLspServer.GetLocations("caret").Single();
+
+            var expectedText = EditorConfigSettingsData.InsertFinalNewLine.GetSettingNameDocumentation();
+
+            var results = await HoverTests.RunGetHoverAsync(
+                testLspServer,
+                expectedLocation).ConfigureAwait(false);
+            Assert.Equal("Insert Final Newline", results.Contents.Third.Value);
```

Comment:
these tests should use the resource string to assert equality.  We do have some spanish tests that run somewhere that would fail if we assert the english version as expected